main/ui/src/main/java/org/cryptomator/ui/unlock/UnlockWorkflow.java

```diff
@@ -73,6 +73,20 @@
 		this.successScene = successScene;
 		this.invalidMountPointScene = invalidMountPointScene;
 		this.errorComponent = errorComponent;
+
+		setOnScheduled(event -> vault.setState(VaultState.PROCESSING));
+		setOnSucceeded(event -> vault.setState(VaultState.UNLOCKED));
+		setOnCancelled(event -> vault.setState(VaultState.LOCKED));
+
+		setOnFailed(event -> {
+			vault.setState(VaultState.LOCKED);
```

Comment:
Same for failed: Set the state in `failed()` method, handle the error in `setOnFailed(event -> {...})`. That way the caller could still decide to apply its own error handling mechanism, but the vault state is still valid.

Reply:
Resolved by 0acc8ddb0a667930a98caacac4ecac8426a83c53.